public/pages/InflightQueries/InflightQueries.tsx

```
@@ -0,0 +1,650 @@
+      value, // Use 'value' instead of 'b' for clarity
+    }));
+  };
+
+  const getChartSpec = (type: string, chartType: 'node' | 'index'): VisualizationSpec => {
+    const isDonut = type.includes('donut');
+
+    return {
+      width: 400,
+      height: 300,
+      mark: isDonut ? { type: 'arc', innerRadius: 50 } : { type: 'bar' },
+      encoding: isDonut
+        ? {
+            theta: { field: 'value', type: 'quantitative' },
+            color: {
+              field: 'label',
+              type: 'nominal',
+              title: chartType === 'node' ? 'Nodes' : 'Indices',
+            },
+            tooltip: [
+              { field: 'label', type: 'nominal', title: chartType === 'node' ? 'Node' : 'Index' },
+              { field: 'value', type: 'quantitative', title: 'Count' },
+            ],
+          }
+        : {
+            x: {
+              field: 'label',
+              type: 'nominal',
+              axis: { labelAngle: -45, title: chartType === 'node' ? 'Node' : 'Index' },
+            },
+            y: {
+              field: 'value',
+              type: 'quantitative',
+              axis: { title: 'Count' },
+            },
+            color: {
+              field: 'label',
+              type: 'nominal',
+              title: chartType === 'node' ? 'Node' : 'Index',
+            },
+            tooltip: [
+              { field: 'label', type: 'nominal', title: chartType === 'node' ? 'Node' : 'Index' },
+              { field: 'value', type: 'quantitative', title: 'Count' },
+            ],
+          },
+    };
+  };
+
+  // Update the embed calls
+  useEffect(() => {
+    if (chartRefByNode.current) {
+      embed(
+        chartRefByNode.current,
+        {
+          ...getChartSpec(selectedChartIdByNode, 'node'),
+          data: { values: getChartData(nodeCounts, 'node') },
+        },
+        { actions: false, renderer: 'svg' }
+      ).catch(console.error);
```

We are silently failing here, could we add a task to dive into what are the possible failures here? we should handle errors more carefully and see if there are any failres we should expose to the user for them to resolve, something like
```
.catch((error) => {
  console.error('Chart rendering failed:', error);
  setChartError('Failed to render chart');
  core.notifications.toasts.addError({...});
});
```

---

Updated